Lesson34/selcan/lesson34-app/src/components/UserDetailsForm/index.jsx

```diff
@@ -0,0 +1,25 @@
+import { useState } from "react";
+import "./styles.css";
+
+// TODO: Update component so that it's possible to change user name and save new user name in the UserContext
+
+export const UserDetailsForm = ({ incomingUsername }) => {
+  const [username, setUsername] = useState("");
+
+  function handleSubmit(e) {
+    e.preventDefault();
```

Comment:
Here, when a username is updated, you need to also update information in the UserContext store (use update action type for it).